hedera-node/docs/design/app/workflows.md

```diff
@@ -1,11 +1,11 @@
-### Workflow Onset
+## Workflow Onset
 
 At the beginning of each workflow, a `Transaction` has to be parsed and validated.
 As these steps are the same for all workflows, this functionality has been extracted and made available independently.
 All related classes can be found in the package `com.hedera.node.app.workflows.onset`.
 Details about the required pre-checks can be found [here](transaction-prechecks.md).
```

Comment:
Neither this document nor `transaction-prechecks.md` uses the word "pure" which is part of the name of the method `pureChecks` implemented by each service.  It would be helpful to relate "pure" to "pre-checks" in both documents.

(Didn't we recently have a confusion where `pureChecks` is  a) run during ingest and b) at least one service (I forget which) did a check which in fact wasn't supposed to happen during ingest (where the tx is rejected without being paid for by the erroring user?  Having "pure" documented here might help prevent that in future.

Reply:
Thanks @david-bakin-sl . `pureChecks` is supposed to be called only in pre-handle and handle. They are different from `transaction-prechecks` that happen in Ingest.  `pureChecks` are checks that are independent of State or Configuration, so they remain the same when called in pre-handle or handle. Call to `PureChecks` in Ingest workflow will be removed in the future, since it was added only as safety measure before calling `calculateFees`. I will update Ingest Workflow part with that.

Reply:
TY, but also please mention "pure" somewhere so readers know that "pure" = "pre" when they actually look at the code.  I suggest?

Reply:
Yes, added the `pureChecks` step, in Ingest Workflow and Pre-handle. Could you please check if that helps?